### ISSUE TITLE
Defaults Experimental Feature Identifies wrong variable type

### ISSUE THREAD

**Issue by fdmsantos** [hashicorp/terraform#28767]
I have a variable with type map of object. One attribute from the object is optional list of string.
Recurring to Defaults feature i want define this attribute as empty list if not set it, but terraform throws an error saying the attribute should be a string (I expected the attribute should be an list)

```
terraform {
  required_version = "0.15.3"
  experiments      = [module_variable_optional_attrs]
}

variable "config" {
  type = map(object({
    url                = string
    createCertificates = optional(bool)
    certificateDomainName = optional(string)
    certificateAnotherNames = optional(list(string))
  }))
}

locals {
  # Set the Defaults
  config = defaults(var.config, {
    # If "createCertificates" isn't set then it will default to false.
    createCertificates = false
    # If "certificateAnotherNames" isn't set then it will default to empty list.
    certificateAnotherNames = []
  })
}

│ Invalid value for "defaults" parameter: .certificateAnotherNames: invalid
│ default value for string: string required.
```

**Comment by apparentlymart** [comment]
Hi @fdmsantos! Thanks for sending this feedback.

The behavior you observed is what is currently intended, because `defaults` for a list expects a default value _for each element of the list_, so that it's possible to have a list of objects where there are default values for attributes in those nested objects.

However, what you've seen here has been a common confusion and so it's unlikely that the `defaults` function as currently implemented will be stabilized as-is. Once we have some more time to work on this experiment we'll try some other designs to meet that need in a less-confusing manner.

I think your observation here overlaps with what was discussed in #28406, so I'm going to close this one just to consolidate that discussion over there. Thanks again!

**Comment by korinne** [comment]
Hi all, we recently released the [Terraform v1.3 alpha](https://releases.hashicorp.com/terraform/1.3.0-alpha20220608), which includes the ability to mark object type attributes as optional and set default values.

I'm following up with issues that have provided feedback on the previous experiment, and wanted to invite you all to try the alpha and provide any feedback on this updated design. You can learn more/add comments[ here](https://discuss.hashicorp.com/t/request-for-feedback-optional-object-type-attributes-with-defaults-in-v1-3-alpha/40550). Thank you so much in advance, and hope you like the feature!
